app-backend/db/src/test/scala/com/azavea/rf/database/ProjectLayerDatasourceDaoSpec.scala

```diff
@@ -30,56 +30,42 @@ class ProjectLayerDatasourceDaoSpec
                                                                     project)
-              datasource <- DatasourceDao.create(
-                dsCreate.toDatasource(dbUser),
-                dbUser
-              )
-              scenesInsert <- (scenes map {
+              datasource <- fixupDatasource(dsCreate, dbUser)
+              dbScenes <- (scenes map {
                 fixupSceneCreate(dbUser, datasource, _)
               }).traverse(
                 (scene: Scene.Create) => SceneDao.insert(scene, dbUser)
               )
-            } yield (scenesInsert, dbUser, dbProject, datasource)
-
-            val datasourceListIO = scenesInsertWithUserProjectIO flatMap {
-              case (
-                  dbScenes: List[Scene.WithRelated],
-                  dbUser: User,
-                  dbProject: Project,
-                  datasource: Datasource
-                  ) => {
-                ProjectDao.addScenesToProject(
-                  dbScenes map { _.id },
-                  dbProject.id,
+              _ <- ProjectDao.addScenesToProject(
+                dbScenes map { _.id },
+                dbProject.id,
+                dbProject.defaultLayerId
+              )
+              layerDatsources <- ProjectLayerDatasourcesDao
+                .listProjectLayerDatasources(
                   dbProject.defaultLayerId
-                ) flatMap { _ =>
-                  {
-                    ProjectLayerDatasourcesDao.listProjectLayerDatasources(
-                      dbProject.defaultLayerId
-                    ) map { datasources: List[Datasource] =>
-                      (dbScenes.map { _.datasource }, datasources)
-                    }
-                  }
-                }
-              }
-            }
+                )
+            } yield (dbScenes.map(_.datasource.id), layerDatsources.map(_.id))
+
+            val (insertedDatasourceIds, listedDatasourceIds) =
+              xa.use(t => datasourcesIO.transact(t)).unsafeRunSync
 
-            val (insertedDatasources, listedDatasources) =
-              xa.use(t => datasourceListIO.transact(t)).unsafeRunSync
-            val insertedIds = insertedDatasources.toSet map {
-              (datasource: Datasource.Thin) =>
-                datasource.id
-            }
-            val listedIds = listedDatasources.toSet map {
-              (datasource: Datasource) =>
-                datasource.id
-            }
-            insertedIds == listedIds
+            println("insertedDatasourceIds")
+            println(insertedDatasourceIds)
+            println("listedDatasourceIds")
+            println(listedDatasourceIds)
+            assert(
+              insertedDatasourceIds.toSet == listedDatasourceIds.toSet,
+              "Listed datasources should be the same as that of scenes in this layer")
+            assert(
+              insertedDatasourceIds.toSet.size == listedDatasourceIds.length,
```

Comment:
If the expectation here is `== 1`, it should be explicit instead of based on a size calculation